extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/AbstractGuacamoleTunnelService.java

```diff
@@ -416,6 +462,21 @@ private GuacamoleTunnel assignGuacamoleTunnel(ActiveConnectionRecord activeConne
             GuacamoleClientInformation info, Map<String, String> tokens,
             boolean interceptErrors) throws GuacamoleException {
 
+        // Set up JDBC-specific tokens
+        tokens.put(JDBC_DATE_TOKEN,
+                new SimpleDateFormat(JDBC_DATE_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
+        tokens.put(JDBC_TIME_TOKEN,
+                new SimpleDateFormat(JDBC_TIME_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
```

Comment:
Did we already go through whether it would be better to overwrite `GUAC_DATE` / `GUAC_TIME` with values that will match the history entry? I remember there being a discussion about that at some point, though I don't recall the consensus.

Reply:
I don't think we've had a discussion about it, no. I went along the assumption that there may be some reason why you wouldn't want to lose the value of `GUAC_DATE` and/or `GUAC_TIME`, but I've no strong attachment to it. I'm happy to just overwrite it, as well, if there aren't any concerns from anyone about that.

Reply:
@mike-jumper Further thoughts on keeping JDBC-specific ones or overwriting the `GUAC_DATE`/`GUAC_TIME` tokens?

Reply:
I think I'm good with keeping things specific to `GUAC_JDBC_*` here, particularly given the expected complexity of the generalized approach. The `GUAC_JDBC_*` tokens are reasonable as they stand and will scratch the community's itch.